Tribler/Core/Session.py

```diff
@@ -601,6 +601,20 @@ def save_collected_torrent(self, infohash, data):
             raise OperationNotEnabledByConfigurationException("torrent_store is not enabled")
         self.lm.torrent_store.put(hexlify(infohash), data)
 
+    def delete_collected_torrent(self, infohash):
+        """
+        Deletes the given torrent into the torrent_store database.
+        :param infohash: The given infohash binary.
+        """
+        if not self.get_torrent_store():
+            raise OperationNotEnabledByConfigurationException("torrent_store is not enabled")
+
+        if self.lm.torrent_store is not None:
+            del self.lm.torrent_store[hexlify(infohash)]
+        else:
+            self._logger.error("Torrent store is enabled but could not delete infohash %s" % infohash)
+            raise RuntimeError("Torrent store is enabled but None")
```

Comment:
No need for this kind of checks, it will fail anyway. just leave the del statement on this chunk.